### PR TITLE
chore: make buffrs project a workspace

### PR DESCRIPTION
**Pull request by tomkarw**
I overcame the temptation to do `[workspace.dependencies]` and all that, it could be a follow up.

### CODE REVIEW

**Review by BlowaXD — APPROVED** [review]

**Comment by mara-schulke** [comment]
Hi, this breaks installation (`cargo install https://github.com/helsing-ai/buffrs) and CI, we opted to not do this a while ago for these reasons. Im trying to look up the old discussion about this.

If you can fix the broken installation + CI its fine / an improvement imo!

File: .github/workflows/cli.yml [hunk 1]
```diff
@@ -31,6 +31,8 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - run: rustup update && rustup component add clippy
+      - name: Install protoc
+        run: sudo apt update && sudo apt install -y protobuf-compiler
```
Comment (BlowaXD) [inline .github/workflows/cli.yml:35]:
```suggestion
      - name: Install Protoc
        uses: arduino/setup-protoc@v2
```

**Comment by tomkarw** [comment]
> cargo install https://github.com/helsing-ai/buffrs

Do you mean that you need to specify the binary? This is already the case, and it's not broken, you just need to specify the binary.

```bash
➜  ~ cargo install --git https://github.com/helsing-ai/buffrs
    Updating git repository `https://github.com/helsing-ai/buffrs`
error: multiple packages with binaries found: buffrs, buffrs-registry. When installing a git repository, cargo will always search the entire repo for any Cargo.toml.
Please specify a package, e.g. `cargo install --git https://github.com/helsing-ai/buffrs buffrs`.
```

 And that's only for installs from repo (which I don't think we have to "support"), once published `cargo install buffrs` and `cargo install buffrs-registry` will work fine.

**Comment by tomkarw** [comment]
Something's wrong with Github CI, I'm seeing spurious 500 errors. I'll come back to it tomorrow.

**Review by mara-schulke — APPROVED** [review]

**Pull request merged**